### PR TITLE
Enable http redirect

### PR DESCRIPTION
**Pull request by juliemar**
### Description
When the client is in production with HTTPS and access the matrix in HTTP the users receive google login error. This PR enables to define the Environment Variable with the correct production protocol 

### How to test?
Folow the this steps: https://github.com/ResultadosDigitais/matrix/issues/142

### Expected behavior

If the environment variable is defined https and the user access http we will redirect him to https

### CODE REVIEW

File: app.server.js [hunk 1]
```diff
@@ -37,6 +39,19 @@ app.use(
 // FIX ME: here we have to get the google APIkey in another way.
 app.locals.googleCredential = new GoogleCredentialController(GOOGLE_CREDENTIAL);
 
+app.use((req, res, next) => {
+  var proto = req.connection.encrypted ? 'https' : 'http';
```
Comment (alesshh) [inline app.server.js]:
```suggestion
  let proto = req.secure ? 'https' : 'http';
```

File: app.server.js [hunk 2]
```diff
@@ -37,6 +39,19 @@ app.use(
...
+  // only do this if you trust the proxy
+  proto = req.headers['x-forwarded-proto'] || proto;
+
+  if (proto !== PRODUCTION_PROTOCOL){
+      res.redirect(`${PRODUCTION_PROTOCOL}://${req.header('host')}${req.url}`)
+  }else{
+    next()
+  }
+})
```
Comment (alesshh) [inline app.server.js]: Suggestion:
I think instead pass a protocol env, users should define a `ENFORCE_SSL` variable:

 ```
if (req.secure) {
    return next()
}

if (process.env.ENFORCE_SSL){
     return res.redirect(`${PRODUCTION_PROTOCOL}://${req.hostname}${req.url}`)
}

next();

File: app.server.js [hunk 3]
```diff
@@ -37,6 +39,19 @@ app.use(
 // FIX ME: here we have to get the google APIkey in another way.
 app.locals.googleCredential = new GoogleCredentialController(GOOGLE_CREDENTIAL);
 
+app.use((req, res, next) => {
+  var proto = req.connection.encrypted ? 'https' : 'http';
+  // only do this if you trust the proxy
+  proto = req.headers['x-forwarded-proto'] || proto;
+
+  if (proto !== PRODUCTION_PROTOCOL){
+      res.redirect(`${PRODUCTION_PROTOCOL}://${req.header('host')}${req.url}`)
```
Comment (alesshh) [inline app.server.js]:
```suggestion
      res.redirect(`${PRODUCTION_PROTOCOL}://${req.hostname}${req.url}`)
```

File: SETUP.md [hunk 4]
```diff
@@ -17,3 +17,28 @@ After repository cloning, run: `npm install`. This will feed all the project nee
 To start: `npm start`
 
 And then open chrome at: `http://localhost:8080`
+
+
+ ## Running with production
+
+1) Configure `GOOGLE_CREDENTIAL` environment with your google autenticator ID follow [this](https://developers.google.com/identity/sign-in/web/sign-in) steps for create a valid credential.
+2) Configure `PRODUCTION_PROTOCOL=https` to force redirect user when http access
```
Comment (alesshh) [inline SETUP.md]: Add new environment variables at deployment file
https://github.com/ResultadosDigitais/matrix/blob/master/app.json#L8

File: app.server.js (hunk 2 above)
Comment (juliemar) [reply app.server.js]: I updated with your suggestion :)

File: SETUP.md (hunk 4 above)
Comment (juliemar) [reply SETUP.md]: Done!

File: app.server.js [hunk 5]
```diff
@@ -13,6 +13,8 @@ const HOST = "0.0.0.0";
 const GOOGLE_CREDENTIAL =
   process.env.GOOGLE_CREDENTIAL ||
   "990846956506-bfhbjsu4nl5mvlkngr3tsmfcek24e8t8.apps.googleusercontent.com";
+const ENFORCE_SSL = process.env.ENFORCE_SSL || false;
```
Comment (alesshh) [inline app.server.js]: Local testing didn't works, I think you need compare environment variables with strings

```suggestion
const ENFORCE_SSL = process.env.ENFORCE_SSL === 'true';
```

File: app.server.js [hunk 6]
```diff
@@ -37,6 +39,16 @@ app.use(
 // FIX ME: here we have to get the google APIkey in another way.
 app.locals.googleCredential = new GoogleCredentialController(GOOGLE_CREDENTIAL);
 
+app.use((req, res, next) => {
+
+  if (ENFORCE_SSL==true & !req.secure){
```
Comment (alesshh) [inline app.server.js]:
```suggestion
  if (ENFORCE_SSL && !req.secure){
```

File: app.json [hunk 7]
```diff
@@ -8,6 +8,14 @@
...
       "description": "Google Application Credential",
       "value": "xxxxxxx.apps.googleusercontent.com"
+    },
+    "ROOMS_DATA": {
+      "description": "List of avaliable rooms office. Read more in SETUP.md",
+      "value": ""
+    },
+    "ENFORCE_SSL": {
+      "description": "If you running in HTTPS this variable forces redirect to HTTPS when user access with HTTP",
+      "value": "true"
```
Comment (jorgemdnt) [inline app.json]: não faria mais sentido o default ser false, pra facilitar subir o servidor sem `https`? o heroku, por exemplo, sobe um server com https?

Comment (alesshh) [reply app.json]: @jorgemdnt this file is to deploy heroku apps, that have ssl by default.
In my concern, SSL it's a best practice that should be encouraged.

File: app.server.js [hunk 8]
```diff
@@ -13,6 +13,8 @@ const HOST = "0.0.0.0";
 const GOOGLE_CREDENTIAL =
   process.env.GOOGLE_CREDENTIAL ||
   "990846956506-bfhbjsu4nl5mvlkngr3tsmfcek24e8t8.apps.googleusercontent.com";
+const ENFORCE_SSL = process.env.ENFORCE_SSL || 'true';
```
Comment (alesshh) [inline app.server.js]: will broken in development environment 🤔

Comment (juliemar) [reply app.server.js]: yesss I'm so sorry for my mistake. I committed my local test :/

File: app.json (hunk 7 above)
Comment (jorgemdnt) [reply app.json]: I'm really sorry, @juliemar & @alesshh . If that is the case, I got it wrong!

**Pull request merged**